src/components/organisms/SettingsChange.tsx

```diff
@@ -0,0 +1,309 @@
+  deviceData: React.Dispatch<React.SetStateAction<[]>>;
+  setDeviceData: React.Dispatch<React.SetStateAction<[]>>;
+  userData: React.Dispatch<React.SetStateAction<{}>>;
+  setUserData: React.Dispatch<React.SetStateAction<{}>>;
+  isLoading: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const SettingsViewState = { DropDown: 0, Read: 1, Change: 2, Updated: 3 } as const; 
+type SettingViewState = typeof SettingsViewState[keyof typeof SettingsViewState];
+
+export default function SettingsChange({
+  setSettingsView,
+  settingsView,
+  // selectedDeviceName,
+  // selectedDeviceIndex,
+  deviceData,
+  setDeviceData,
+  userData,
+  setUserData,
+  isLoading,
+}: Props) {
+  const submitSettingsChange = () => {
+    let firstName: string = document.getElementById("firstName").value;
+    let lastName: string = document.getElementById("lastName").value;
+    let email: string = document.getElementById("email").value;
+    let phoneNumber: string = document.getElementById("phoneNumber").value;
+    let oldPassword: string = document.getElementById("oldPassword").value;
+    let newPassword: string = document.getElementById("newPassword").value;
+
+    let deviceName: string = document.getElementById("deviceName").value;
+    let hotTempSetting: number =
+      document.getElementById("hotTempSetting").value;
+    let coldTempSetting: number =
+      document.getElementById("coldTempSetting").value;
+    let zipcode: string = document.getElementById("zipcode").value;
+
+    const updatedUserSettings: {} = {
+      // comments contain original data
+      first_name: firstName ? firstName : userData.first_name, // Akemi
+      last_name: lastName ? lastName : userData.last_name, // Kimura
+      email: email ? email : userData.email, //test@test.com
+      phone_number: phoneNumber ? phoneNumber : userData.phone_number, //090-0000-0000
+      old_password: oldPassword ? oldPassword : userData.old_password, //secretPassword
+      new_password: newPassword ? newPassword : "secretPassword",
+    };
+
+    const updatedDeviceSettings: {} = {
+      device_name: deviceName ? deviceName : deviceData[0].device_name, //Roppongi_Device
+      temperature_upper_limit: hotTempSetting
+        ? hotTempSetting
+        : deviceData[0].temperature_upper_limit, //30.2
+      temperature_lower_limit: coldTempSetting
+        ? coldTempSetting
+        : deviceData[0].temperature_lower_limit, //27?
+      zip_code: zipcode ? zipcode : deviceData[0].zip_code, //1001701
+    };
+
+    api.put(`/settings/user/`, updatedUserSettings).then((res) => {
+      console.log(res.data);
+      location.reload();
+    });
+    
+    api.put(`/settings/device/`, updatedDeviceSettings).then((res) => {
+      console.log(res.data);
+    });
+    setSettingsView(SettingsViewState.Updated)
+  };
+
+  
+
+  console.log("CHANGE", setSettingsView);
+  console.log("CHANGE", deviceData);
+
+  return (
+    <>
```

Comment:
Can you wrap whole contents by `<form>`?
ex. 
`<form onSubmit={handleSubmit(signupOnSubmit)}>`

Reply:
I changed it to <form>.   Do I need to add `onSubmit={handleSubmit(signupOnSubmit)}>`?

Reply:
Skipping for now..